### ISSUE TITLE
Bug in resample_from_grid()

### ISSUE THREAD

**Issue by choisy** [choisy/sptools#17]
Here is a demonstration of the bug. For that, we need the following packages:

```
# devtools::install_github("choisy/srtmVN")
# devtools::install_github("choisy/worldpopVN")
library(magrittr)
library(sp)
```

The data we need:

```
country <- sptools::gadm("vietnam", "sp", 0)
elevation <- srtmVN::getsrtm()
popdensity <- worldpopVN::getpop()
```

Let's make a grid:

```
grid <- country %>%
  sptools::make_grid(100) %>%
  sptools::add_from_raster(elevation, "elevation")
```

Let's make a second version of this grid, where we remove the variable `longitude`:

```
grid2 <- grid
grid2@data[, "longitude"] <- NULL
```

So, the only difference between `grid` and `grid2` is that `grid` contains the variable `longitude` that `grid2` does not have. We can check that the coordinates are the same:

```
identical(coordinates(grid), coordinates(grid2))
```

The bug is due to the fact that 

```
sptools::resample_from_grid(popdensity, grid)
```

works just fine whereas 

```
sptools::resample_from_grid(popdensity, grid2)
```
 fails with the following message:

```
Error in .intersectExtent(x, y, validate = TRUE) : 
  Objects do not intersect 
```

**Comment by choisy** [comment]
Note that it fails in the `raster::resample()` call of the `sptools::resample_from_grid()` function.

**Comment by LucieContamin** [comment]
Fixed in pull request #18
